### DISCUSSION TITLE
New user password library installed from 3.1.1.48, impacts?

### DISCUSSION THREAD

**Discussion by nilsteampassnet** [nilsteampassnet/TeamPass#4020] [General]
### What's new?

From version 3.1.1.48, a new library ([Symfony Password](https://symfony.com/doc/current/security/passwords.html)) has been implemented to get rid of the existing one.
This has no impacts on your users as the "migration" of encryption is performed in background during 2 possible steps:
* when user login into Teampass, the passwords is decrypted with "old library" and encrypted back with the new one and stored in DB
* when Admin is changing the user's password

Notice that "old library" will be removed in future release 3.1.3.
In the meantime, it will be required to ensure that all you users have their password updated.

#### In admin page, a new information paragraph
![image](https://github.com/nilsteampassnet/TeamPass/assets/1197546/196ae1b7-272a-424f-aa78-86565aa2953d)

#### In Users list datatable
A new red symbol will be displayed in case the user's password is encrypted with old library
![image](https://github.com/nilsteampassnet/TeamPass/assets/1197546/719ead8c-fff0-46eb-859c-7a9a35801b3f)

#### During Upgrade process
During "Checks" step of upgrade process, you may see a new warning regarding this.
![image](https://github.com/nilsteampassnet/TeamPass/assets/1197546/2997a2e9-2d03-456c-9f55-cb568cc6a4ef)

Indeed in order to upgrade to future release 3.1.3, it will be required to have this done as "old library" will be removed.

**Comment by rodrigues730** [comment]
Ola ! 
Agradeço pelo disponibilidade dessa solução fantastica
gostaria de um apoio depois de vir da versão anterior.
Apos fazer update nenhuma senha abre fica sempre em load intem..
seria algum modulo que esqueci de instalar?


Oracle Linux Server 9.3

PHP version: 8.2.17
Memory limit: 128M
Memory usage: 527.84 KB
Maximum time execution: 30
Maximum file size upload: 10M

Server version: 10.5.24-MariaDB
Database size: 6.08MB





![evidencia_pass](https://github.com/nilsteampassnet/TeamPass/assets/66528152/7fdcf07c-d8da-4f97-a7ee-03546a356906)

**Reply by rodrigues730** [reply]
Identifiquei no log php estes erros pode verificar ?

![evidencia2](https://github.com/nilsteampassnet/TeamPass/assets/66528152/00506bd0-6be3-4d68-87b6-875aa546bdb9)

**Comment by deanbruhn** [comment]
I have an issue where I am being warned that I have 3 users that haven't had their encryption updated, but none of my users are showing the indicator. I am on 3.1.2.45. I believe these deleted users that were removed from the system, but I am not sure what I need to do to fix this before it becomes imperative?

**Reply by nilsteampassnet** [reply]
Can you share a screencpature with this users please?

**Reply by deanbruhn** [reply]
![ScreenShot](https://github.com/user-attachments/assets/7ef5cf96-2ab6-48e6-bf21-2a52f5ca7264)

**Reply by deanbruhn** [reply]
Hey Nils, heads up. This might be related to issue #4229 also. I have been using this system for such a long time, I am wondering if the way user deletion was handled has been changed, and the old ways aren't properly supporting some of the new user functions and cleanup.

**Reply by whosht** [reply]
Same probleme here - see #4634

**Comment by ivancasanovaicp** [comment]
Hello, Same problem here.

I have 26 remaining users. But the user list, no one has the icon.

Thanks.

**Reply by g0lgs** [reply]
My system (3.1.4.19) shows 'Remaining users: 10' but none of them have red symbol most of the (17) users are LDAP enabled (not the admins).

The teampass_users table shows 5 with '_deleted' or similar appended to the original names.

**Reply by jonedved** [reply]
In my case, remaining users number included deleted users.
I had to connect directly to the DB and delete the "%deleted%" users.
Now my status is OK.

**Reply by ivancasanovaicp** [reply]
I just made the same change, and now it works correctly.

Thank you so much for your help.

**Reply by g0lgs** [reply]
I got rid of the 5 "%deleted%" users, but it still reports 5 users using old passwords, but none of them show the red symbol.

**Reply by jonedved** [reply]
Take a look at the users in user table - directly in DB. And compare to the list of users you see in TeamPass admin view. That will probably help you identify which are the culprit.

**Reply by nilsteampassnet** [reply]
Did you enabled ?
![image](https://github.com/user-attachments/assets/aed6a15b-3b02-4b81-90da-b2c38b32abe9)

**Comment by jonedved** [comment]
Hello, would it be possible to add a button or link to save a CSV file with the users that need to take this action? Ideally user name and e-mail address would do the job.
